rasterio/_env.pyx

```diff
@@ -68,11 +68,11 @@ cdef void log_error(CPLErr err_class, int err_no, const char* msg) with gil:
 # other platforms. Each calls log_error().
 IF UNAME_SYSNAME == "Windows":
     cdef void __stdcall logging_error_handler(CPLErr err_class, int err_no,
-                                              const char* msg):
+                                              const char* msg) with gil:
         log_error(err_class, err_no, msg)
 ELSE:
     cdef void logging_error_handler(CPLErr err_class, int err_no,
-                                    const char* msg):
+                                    const char* msg) with gil:
```

Comment:
@jdmcbr I debug the Cython modules with a combination of log messages (the origin of the `log.debug(...)` code sprinkled around rasterio) and lldb on OS X and gdb on Linux. My C debugger skills are pretty modest, but `lldb python` and `run -m pytest` will at least give  the stack where things the failure has occurred and allow upward traversal into the C code generated by Cython, if not the Cython code itself.